backend/controller/seriesController.js

```diff
@@ -0,0 +1,309 @@
+const asyncHandler = require("../middleware/asyncHandler");
+const seriesModel = require("../model/seriesSchema");
+const CustomError = require("../utils/customError");
+
+const cloudinaryFileUpload = require("../utils/fileUpload.cloudinary");
+const { cloudinaryFileDelete } = require("../utils/fileDelete.cloudinary");
+
+const likeAndDislike = require("../utils/likeDislike");
+
+/********************
+ * @httpCreateSeries
+ * @route http://localhost:8081/api/v1/series/
+ * @description  controller to create the series
+ * @parameters {request body object}
+ * @return { Object } series object
+ ********************/
+const httpCreateSeries = asyncHandler(async (req, res, next) => {
+  // get require field
+  const {
+    seriesTitle,
+    seriesSummary,
+    genres,
+    releaseDate,
+    cast,
+    rating,
+    language,
+    director,
+    originCountry,
+  } = req.body;
+
+  let seriesDetails = {
+    seriesTitle,
+    seriesSummary,
+    genres,
+    releaseDate,
+    cast,
+    rating,
+    language,
+    director,
+    originCountry,
+    //default thumbnail value
+    thumbnail: [
+      {
+        thumbnailUrl:
+          "https://res.cloudinary.com/dp3qsxfn5/image/upload/v1687258494/default_thumbnail_mi4zwc.webp",
+      },
+    ],
+
+    // default trailer value
+    trailer: [
+      {
+        trailerUrl:
+          "https://res.cloudinary.com/dp3qsxfn5/video/upload/v1687258296/Default_video_ikitm6.mp4",
+      },
+    ],
+  };
+
+  // checking for missing fields
+  const requiredFields = [
+    "seriesTitle",
+    "seriesSummary",
+    "genres",
+    "releaseDate",
+    "cast",
+    "rating",
+    "language",
+    "director",
+    "originCountry",
+  ];
+  const missingRequiredFields = requiredFields.filter(
+    (field) => !seriesDetails[field]
+  );
+  if (missingRequiredFields.length > 0) {
+    // retrieve first missing fields and send the message
+    const missingField = missingRequiredFields[0];
+    return next(new CustomError(`Missing Field - ${missingField}`));
+  }
+
+  // create mongoose
+  const seriesObject = seriesModel(seriesDetails);
+  const seriesData = await seriesObject.save();
+
+  // check for seriesFiles
+  if (!seriesData) {
+    return next(
+      new CustomError(
+        "Series data fail to save to database! Please try again",
+        400
```

Comment:
This is a server error and should be 5xx.

Reply:
this file is removed.  the seriesController.js is combined with content